### PR TITLE
add progress bar

### PR DESCRIPTION
**Pull request by apawlik96**

### CODE REVIEW

File: src/containers/ToDoList/ToDoList.jsx [hunk 1]
```diff
@@ -151,6 +156,26 @@ export const ToDoList = () => {
...
+            <progress
+              value={tasks.filter(taskFilters[FilterType.COMPLETED]).length}
+              max={tasks.length}
+              style={{ width: "100%" }}
+            />
+            <div className="progress-number">
+              {Math.round(
+                (tasks.length > 0 &&
+                  tasks.filter(taskFilters[FilterType.COMPLETED]).length /
+                    tasks.length) * 100
```
Comment (przemek-ycd) [inline src/containers/ToDoList/ToDoList.jsx]:
```suggestion
                (hasTasks && numberOfTasks / tasks.length) * 100
```

File: src/containers/ToDoList/ToDoList.jsx [hunk 2]
```diff
@@ -71,9 +73,12 @@ export const ToDoList = () => {
   }, []);
 
   const handleTaskMark = (id) => {
-    const newTasksList = tasks.map((task) =>
+    const updatedTasks = tasks.map((task) =>
       task.id === id ? { ...task, checked: !task.checked } : task
     );
+
+    const newTasksList = updatedTasks.sort((a, b) => a.checked - b.checked);
```
Comment (przemek-ycd) [inline src/containers/ToDoList/ToDoList.jsx]: Operations like filtering and/or sorting should be performed while reading the state, not while writing.

File: src/containers/ToDoList/ToDoList.jsx [hunk 3]
```diff
@@ -142,6 +147,11 @@ export const ToDoList = () => {
     setCurrentSort(currentSort === "newestTask" ? "oldestTask" : "newestTask");
   };
 
+  const numberOfCompletedTasks = tasks.filter(
+    taskFilters[FilterType.COMPLETED]
+  ).length;
+  const totalTasksCount = tasks.length;
```
Comment (przemek-ycd) [inline src/containers/ToDoList/ToDoList.jsx:156]: Yeah, this is great!

File: src/containers/ToDoList/ToDoList.jsx [hunk 4]
```diff
@@ -151,6 +161,25 @@ export const ToDoList = () => {
...
         </StyledWrapperTitle>
+
+        <StyledWrapperCompletedTasks>
+          <span>Completed tasks</span>
+          <div className="progress-bar">
+            <progress
+              value={numberOfCompletedTasks}
+              max={totalTasksCount}
+              style={{ width: "100%" }}
+            />
```
Comment (przemek-ycd) [inline src/containers/ToDoList/ToDoList.jsx]: Why not use the `ProgressBar` from the MUI? :-)
https://mui.com/material-ui/react-progress/#linear

File: src/containers/ToDoList/ToDoList.jsx [hunk 5]
```diff
@@ -173,16 +202,24 @@ export const ToDoList = () => {
...
+                  id={task.id}
+                  text={task.text}
+                  checked={task.checked}
+                  onCheck={handleTaskMark}
+                  onDelete={handleDeleteTask}
+                  dateAdded={task.dateAdded}
+                />
+              </Reorder.Item>
+            ))}
+          </Reorder.Group>
```
Comment (przemek-ycd) [inline src/containers/ToDoList/ToDoList.jsx:237]: That all is responsible for ensuring the possibility of re-ordering the elements -- that right?

**Review by przemek-ycd — COMMENTED** [review]
And two tiny comments for the future:
1. PR titles and commit messages should be written in imperative form, starting with capital letter:
    * `Add new feature` - is ok
    * `added new feature` - is not ok
    * `Adding new feature` - is not ok
2. This PR contains both reordering and progress bar - but don't change this now ;-)

File: src/containers/ToDoList/ToDoList.jsx (hunk 5 above)
Comment (apawlik96) [reply src/containers/ToDoList/ToDoList.jsx:237]: Yes, both: the reordering and the slide effect when a task is completed/marked

File: src/containers/ToDoList/ToDoList.jsx [hunk 6]
```diff
@@ -151,6 +166,39 @@ export const ToDoList = () => {
...
+                    ? (numberOfCompletedTasks / totalTasksCount) * 100
+                    : 0
+                }
+                sx={{
+                  width: "100%",
+                  "& .MuiLinearProgress-bar": {
+                    backgroundColor: "#2e9acc",
+                  },
+                  backgroundColor: "#b4b4b8",
+                }}
```
Comment (przemek-ycd) [inline src/containers/ToDoList/ToDoList.jsx]: This should go to const `progressBarStyles` not to mess with the JSX part of the file. In other words - for better readability

File: src/containers/ToDoList/ToDoList.jsx [hunk 7]
```diff
@@ -151,6 +166,39 @@ export const ToDoList = () => {
...
+              />
+            </Box>
+            <Box>
+              <Typography>
+                {`${Math.round(
+                  (totalTasksCount > 0 &&
+                    numberOfCompletedTasks / totalTasksCount) * 100
+                )}%`}
```
Comment (przemek-ycd) [inline src/containers/ToDoList/ToDoList.jsx]:
```suggestion
                {Math.round(
                  (totalTasksCount > 0 &&
                    numberOfCompletedTasks / totalTasksCount) * 100
                )}%
```

**Review by przemek-ycd — COMMENTED** [review]
Just couple of minor comments, but it looks good now already :-)

**Review by przemek-ycd — APPROVED** [review]

**Pull request merged**